src/AgentCommerce/Routing/AgentRequestContextResolver.php

```diff
@@ -75,9 +87,32 @@ public function resolve(Request $request): void
         }
 
         $source = $this->resolveContextSource($token);
+
+        $criteria = new Criteria();
+        $criteria->addFilter(
+            new EqualsFilter('storefrontSalesChannel.active', true),
+            new EqualsFilter('salesChannel.active', true),
+            new EqualsFilter('salesChannel.typeId', SwagPayPal::SALES_CHANNEL_TYPE_AGENT_COMMERCE),
+            new EqualsFilter('salesChannel.id', $source->salesChannelId),
+        );
+
         $context = new Context($source);
+        $productExport = $this->productExportRepository->search($criteria, $context)->first();
+        if (!$productExport) {
+            throw AgentException::unauthorized('Sales channel not found');
+        }
+
+        preg_match('/CART-([a-zA-Z0-9]{32})/', $request->getPathInfo(), $matches);
```

Comment:
Nothing in shopware ensures that a token shaped that way:
```suggestion
        preg_match('/CART-(\w+)/', $request->getPathInfo(), $matches);
```

Reply:
done in shopware/SwagPayPal#324